servers/rendering/rendering_server_default.h

```diff
@@ -78,7 +78,7 @@ class RenderingServerDefault : public RenderingServer {
 	static void _thread_callback(void *_instance);
 	void _thread_loop();
 
-	Thread::ID server_thread;
+	Thread::ID server_thread = Thread::get_caller_id();
```

Comment:
Is this safe to assign like this? @RandomShaper

Reply:
Some servers (including this) have this:
```
	if (!p_create_thread) {
		server_thread = Thread::get_caller_id();
	} else {
		server_thread = 0;
	}
```

I'd say it'd be better to keep the `server_thread = 0;` in the header and remove the `else` in these cases.

Reply:
That's a good point. Those `if`s should be replaced by just `server_thread = Thread::get_caller_id()` (or, alternatively, `Thread::get_main_id()`). Regardless, `p_create_thread`, at that point the "server thread" is the main one. The `0` would be only interesting as a default for initialization to avoid the garbage until something more sensible can be assigned. Later, the `thread_loop` functions will reassign `server_thread` and, that time, if the server is actually running on a non-main thread, the right value will be collected there.

Reply:
I just realized that `server_thread` had garbage data in it because I skipped the initialization. It was inside a `THREADS_ENABLED`.